Add adapter for converting LORIS username/password to HTTP Bearer token

driusan opened the pull request:
Added an adapter which converts usernames and passwords to tokens via the LORIS API.

This is not thread-safe, since it uses a hack of adding url as a property to the _loris_adapter function in order to pass the URL between the composite credential and the adapter.

driusan committed:
Add adapter for converting LORIS username/password to HTTP Bearer token

yarikoptic commented:
Let me digest it later while looking at the code not in the phone. May be I will see where in the interactions credential could become aware of URL

driusan commented:
Moving the adapters from bare functions to class methods would make sense to me, but I don't know anything about the other NDA adapter and couldn't test it if I changed that part of the architecture

yarikoptic commented:
NDA adapter isn't used in production yet, was more of an unfinished hackathon effort,,, so it would be ok to do what needs to be done

driusan committed:
Pass CompositeCredential to adapters that it uses

This allows features of the Composite (such as the URL) to be accessed by the adapters

driusan commented:
I realized that it doesn't need to be a method, it just needs to be passed the parent class instance, so I did that instead.. seems more reasonable to me now.

yarikoptic commented on datalad/support/third/loris_token_generator.py:
```diff
@@ -0,0 +1,23 @@
+import sys
```
Please borrow the header from other files, or add a custom one (about license/copyright, if needed)

yarikoptic reviewed (approved):
Lovely, especially if it works ;-)

In principle/ideally there should be a test for the interaction.  httpretty might come handy - there are some tests using it.  Since otherwise I am afraid we might break it sometime without realizing that

driusan commented:
squashed and pushed directly to the branch. Still needs test.
